### PR TITLE
fix: save tagged video frames as "jpg" files

### PR DESCRIPTION
**Pull request by elizabethhalper**
Video frames were not correctly exported when using Pascal VOC. Images were generated but did't have "jpg" file extension; it is expected that all video frames are exported to JPG images when project is exported

### CODE REVIEW

File: src/providers/export/tensorFlowPascalVOC.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export class TFPascalVOCJsonExportProvider extends ExportProvider<ITFPascalVOCJs
     }
 
     private async exportSingleImage(jpegImagesFolderName: string, element: IAssetMetadata): Promise<void> {
-        const imageFileName = `${jpegImagesFolderName}/${element.asset.name}`;
+        const imageFileName = `${jpegImagesFolderName}/${element.asset.name.replace(/\.[^/.]+$/, ".jpg")}`;
```
Comment (wbreza) [inline src/providers/export/tensorFlowPascalVOC.ts]: Agreed - need some more tests.  How can we validate that this happens consistently across providers?

Comment (elizabethhalper) [reply src/providers/export/tensorFlowPascalVOC.ts]: @wbreza I agree, but I'm not sure how to do this given the tests we have are already "testing" that assets are exported to "jpg" images. Let's have a discussion offline as to how to refactor them to test the correct thing.

File: src/common/utils.test.ts [hunk 2]
```diff
@@ -50,5 +50,12 @@ describe("Helper functions", () => {
 
             expect(decryptedProject).toEqual(testProject);
         });
+
+        it("Replaces an extension with new extension", () => {
+            expect(replaceExtension("filename.badext", ".jpg")).toEqual("filename.jpg");
+            expect(replaceExtension("filename.this.is.bad", ".jpg")).toEqual("filename.this.is.jpg");
+            expect(replaceExtension("filename", ".jpg")).toEqual("filename");
```
Comment (mydiemho) [inline src/common/utils.test.ts]: if a file has no extension, should we add an extension?

File: src/common/htmlFileReader.test.ts [hunk 3]
```diff
@@ -3,8 +3,26 @@ import HtmlFileReader from "./htmlFileReader";
...
 import MockFactory from "./mockFactory";
+import { AssetState, IAsset } from "../models/applicationState";
 
 describe("Html File Reader", () => {
+    const assetTestCache = new Map<string, IAsset>();
+
+    beforeEach(() => {
+        assetTestCache.clear();
+
+        document.createElement = jest.fn((elementType) => {
```
Comment (wbreza) [inline src/common/htmlFileReader.test.ts]: Refactored html element mocks.

File: src/common/htmlFileReader.ts [hunk 4]
```diff
@@ -104,12 +104,21 @@ export default class HtmlFileReader {
...
+     * Extracts the specified image frame from a video asset
+     * @param asset The asset video frame to retrieve from the parent video
+     */
+    public static async getAssetFrameImage(asset: IAsset): Promise<Blob> {
         return new Promise((resolve, reject) => {
             const cachingEnabled = false;
             let refresh = !cachingEnabled;
             let video: HTMLVideoElement = this.videoAssetFiles[asset.parent.name];
 
+            // Ensure the asset name includes jpg file extension
```
Comment (wbreza) [inline src/common/htmlFileReader.ts]: Moving fix into HtmlFileReader to easily apply to all export providers.

File: src/providers/export/vottJson.ts [hunk 5]
```diff
@@ -32,7 +32,7 @@ export class VottJsonExportProvider extends ExportProvider<IVottJsonExportOption
         await results.forEachAsync(async (assetMetadata) => {
             return new Promise<void>(async (resolve) => {
                 const blob = await HtmlFileReader.getAssetBlob(assetMetadata.asset);
-                const assetFilePath = `vott-json-export/${assetMetadata.asset.id}.jpg`;
+                const assetFilePath = `vott-json-export/${assetMetadata.asset.name}`;
```
Comment (wbreza) [inline src/providers/export/vottJson.ts]: Normalized VoTT json to follow same convensions as other export providers that same image binaries.

File: src/providers/export/tensorFlowPascalVOC.ts (hunk 1 above)
Comment (wbreza) [reply src/providers/export/tensorFlowPascalVOC.ts]: Moved fix into HtmlFileReader and added tests.

File: src/common/utils.test.ts (hunk 2 above)
Comment (wbreza) [reply src/common/utils.test.ts]: Moved fix into `HtmlFileReader` and added tests.  Will always add "jpg" extension if it does not already exist.

**Pull request merged**